Ibex integration

micprog opened the pull request:
This is an initial placement of the ibex core into the pulp cluster. It should not affect the cluster using riscv core. The hier-icache interface was updated to a commit on the fix_icache branch to allow for 32-bit instructions. Ibex has not been fully tested yet due to the SDK not allowing for this yet, but work is in progress.

accuminium commented:
Thanks, @micprog. Is this PR still a draft or ready for review and potentially merging?

micprog commented:
If ok with you, I suggest review and merging. The changes should not affect the standard configuration, so to avoid large deviation with other developments I would prefer to merge. Further changes should be limited to bugfixes or ibex-specific changes.

OttG commented:
Overall looks good, but I have a couple of question:
- Shouldn't we merge the icache-fix branch into the master before using that commit? @accuminium 
- I would like the instance name of the cores to have different name depending on the version (e.g. RISCV_CORE, IBEX_CORE). I mean it's not a big deal but could be easier to write general tcl scripts for waves debugging.

micprog commented:
Thanks @OttG for the edits.
- Concerning icache-fix, I assume you mean the version of hier-icache used? Previous versions of hier-icache used were also from branches away from master, but I updated it to v1.2.0, a new commit on master with up-to-date changes.
- Concerning naming of the cluster cores, I renamed to RISCV_CORE and IBEX_CORE, makes sense to differentiate these.
